File: src/main/kotlin/side/project/checkgeom_severless/domain/MoreView.java
package side.project.checkgeom_severless.domain;

public record MoreView (
       boolean moreView,
       int totalCount
){
    public static MoreView create (
            int totalCount
    ) {
        return new MoreView(
                isMoreView(totalCount),
                totalCount
        );
    }

    private static boolean isMoreView(int totalCount) {
        return totalCount >= 10;
    }

}

File: src/main/kotlin/side/project/checkgeom_severless/domain/MoreView.kt
package side.project.checkgeom_severless.domain

@JvmRecord
data class MoreView(
    val moreView: Boolean,
    val totalCount: Int
) {
    companion object {
        fun create(
            totalCount: Int
        ): MoreView {
            return MoreView(
                isMoreView(totalCount),
                totalCount
            )
        }

        private fun isMoreView(totalCount: Int): Boolean {
            return totalCount >= 10
        }
    }
}
